File: 项目/OC项目/斗鱼(OC)/斗鱼(OC)/Classes/Home/View/MoreButton.m
//
//  MoreButton.m
//  斗鱼(OC)
//
//  Created by 金亮齐 on 2017/6/5.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//
#import "MoreButton.h"

@implementation MoreButton

- (instancetype)init {
    if (self = [super init]) {
        self.titleLabel.textAlignment = NSTextAlignmentRight;
        self.titleLabel.font = [UIFont systemFontOfSize:13];
        [self setTitleColor:[UIColor grayColor] forState:UIControlStateNormal];
        self.imageView.contentMode = UIViewContentModeCenter;
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];
    
    self.titleLabel.x = 0;
    self.titleLabel.width = self.width * 0.7;
    self.titleLabel.y = 0;
    self.titleLabel.height = self.height;
    
    self.imageView.x = self.titleLabel.width;
    self.imageView.y = 0;
    self.imageView.width = self.width - self.imageView.x;
    self.imageView.height = self.height;
}


@end

File: 项目/Swift项目/小黄车(Swift)/小黄车(Swift)/Classes/Extension/MoreButton.swift
//
//  MoreButton.swift
//  小黄车(Swift)
//
//  Created by 金亮齐 on 2017/5/31.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//

import UIKit

class MoreButton: UIButton {
    override init(frame: CGRect) {
        super.init(frame: frame)
        // 设置字体颜色
        self.setTitleColor(UIColor.colorWithCustom(r: 10, g: 10, b: 10), for: .normal)
        self.titleLabel?.font = UIFont.systemFont(ofSize: 16)
        self.imageView?.contentMode = .scaleAspectFit
        self.titleLabel?.adjustsFontSizeToFitWidth = true
        // 让文字居中显示
        self.titleLabel?.textAlignment = NSTextAlignment.center;
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        self.imageView?.mg_y = self.mg_height * 0.15;
        self.imageView?.mg_centerX = self.mg_width * 0.5;
        
        // 调整文字位置
        self.titleLabel?.mg_width = self.mg_width;
        self.titleLabel?.mg_height = self.mg_height - self.imageView!.mg_height;
        self.titleLabel?.mg_x = 0;
        self.titleLabel?.mg_y = self.imageView!.frame.maxY
    }
}
